### PR TITLE
Restructure unitary system get input

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #ISSUENUMBERHERE (IF THIS IS A DEFECT)
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh]: I have to figure out how to get this function back in. Temporarily commented out (and pasted the little chunk of code in 3 places) to test for diffs.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh]: Moved from here up as member. This line will get deleted.

File: src/EnergyPlus/UnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:7126]: Moved input processing to processInputSpec. This should be able to be called by alternate getInput functions.

**Comment by rraustad** [comment]
@Myoldmopar @mjwitte @mitchute @jcyuan2020 the CI is now green. This is the first step to allow the UnitarySystem to model other system types (e.g., HVACDXSystem's CoilSystem objects). Once this is reviewed and merged, code that models other system types can be added to UnitarySystem to allow detailed testing of all possible coil types and configurations.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:188]: I added this as a member to UnitarySystem but never used it.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:522]: Split getUnitarySystemInputData into 2 parts. The processInputSpec function processes all inputs based on the data in input_data. Then it returns and finishes processing. The setupOutputVariable calls are in getUnitarySystemInputData since the report variables for different models may be different from UnitarySystem.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:524]: moved to member function.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:881]: Shoot, should have deleted this.

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:403]: variable used to hold off on call to setupOutputVariable until processing is complete. Not positively sure this will avoid duplicate calls but it fixed initial problems and matches what develop is doing (based on CI results).

File: src/EnergyPlus/UnitarySystem.hh
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.hh:128]: new struct to hold UnitarySystem model inputs. This array will be used to translate inputs for other object types to comparable UnitarySystem inputs.

File: src/EnergyPlus/UnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:124]: The basic change here was to split getUnitarySystemInputData into 2 parts. A more lean getUnitarySystemInputData specific to UnitarySystem and it's report variables and a second function, processInputData, that can be used by other code (e.g., a getInput for DXSystem's CoilSystem model). Code duplication in the future should be minimized.

File: src/EnergyPlus/UnitarySystem.hh
Comment (Myoldmopar) [reply src/EnergyPlus/UnitarySystem.hh:522]: :+1: I like where this is heading!

File: src/EnergyPlus/UnitarySystem.hh
Comment (Myoldmopar) [reply src/EnergyPlus/UnitarySystem.hh:188]: Good cleanup, we should go in and clean out any unused member variables across the whole codebase.

File: src/EnergyPlus/UnitarySystem.hh
Comment (Myoldmopar) [reply src/EnergyPlus/UnitarySystem.hh:128]: OK, so this is essentially just a list of the input fields that unitary uses.  I like having a standardized signature that other parent objects can use as we adapt them.  This is really good.  For the record you don't need to explicitly list out the empty constructor and destructors for classes like this that won't be involved in inheritance and the like.

File: src/EnergyPlus/UnitarySystem.cc
Comment (Myoldmopar) [reply src/EnergyPlus/UnitarySystem.cc:124]: :+1:

File: src/EnergyPlus/UnitarySystem.cc
Comment (Myoldmopar) [inline src/EnergyPlus/UnitarySystem.cc:2916]: It would be really good to go ahead and put braces around this call to GetOnlySingleNode even though it's not syntactically necessary.  Extra indentation can cause weirdness.  Nothing to necessarily change right now though.

File: src/EnergyPlus/UnitarySystem.cc
Comment (Myoldmopar) [inline src/EnergyPlus/UnitarySystem.cc:3016]: Are you just reading these off for convenience?

File: src/EnergyPlus/UnitarySystem.cc
Comment (Myoldmopar) [inline src/EnergyPlus/UnitarySystem.cc:16897]: All the changes in here are what I would be expecting.  Lots and lots of simply rewiring the input routine to get data from a structure rather than from the input processor.  This is going to make generalizing the unitary system a lot nicer!

**Review by Myoldmopar — APPROVED** [review]
I'll be leaning on the entirety of unitary unit tests and example files to validate these changes.  This is really good stuff though. I had a couple questions, but assuming CI is OK, those questions don't need to hold up the PR.

**Comment by Myoldmopar** [comment]
And yes, that is a very green CI result set!  Going in.  Thanks @rraustad!

File: src/EnergyPlus/UnitarySystem.cc
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:3016]: Yes, this was for speed of implementation.

**Pull request merged**
